shapes/Rectangle.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include "GeometricShape.hpp"
+
+namespace jbrush
+{
+class Rectangle final : public GeometricShape
+{
+public:
+    /* Constructors */
+    Rectangle(const QPoint& topL, const QPoint& bottomR) : Rectangle(topL, bottomR, DEFAULT_GEOMETRIC_PROPS) {}
+    Rectangle(const QPoint& topL, int width, int height) : Rectangle(topL, width, height, DEFAULT_GEOMETRIC_PROPS) {}
+    Rectangle(int x, int y, int width, int height) : Rectangle(x, y, width, height, DEFAULT_GEOMETRIC_PROPS) {}
+    Rectangle(const QPoint& topL, const QPoint& bottomR, GeometricShapeProperties);
```

Comment:
Let's just make them actual default parameters then if it's going to look like this.
```suggestion
    Rectangle(const QPoint& topL, const QPoint& bottomR, GeometricShapeProperties = DEFAULT_GEOMETRIC_PROPS);
```